### ISSUE TITLE
what's wrong with my dataIter?why I always get the "Resetting Data Iterator" and get the same accuracy

### ISSUE THREAD

**Issue by lugui2009** [apache/incubator-mxnet#2640]
Hi,everyone .I write an DataIter of my own to test my network.However,when I take into the model.fit(),I get the message "Resetting Data Iterator",then get the same Train-accuracy every epoch.Belows are the messages:
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
INFO:root:Epoch[26] Batch [200] Speed: 2827.38 samples/sec      Train-accuracy=0.007743
INFO:root:Epoch[26] Batch [400] Speed: 2802.66 samples/sec      Train-accuracy=0.008020
INFO:root:Epoch[26] Resetting Data Iterator
INFO:root:Epoch[26] Time cost=7.802
INFO:root:Epoch[27] Batch [200] Speed: 2761.99 samples/sec      Train-accuracy=0.007743
INFO:root:Epoch[27] Batch [400] Speed: 2779.24 samples/sec      Train-accuracy=0.008020
INFO:root:Epoch[27] Resetting Data Iterator
INFO:root:Epoch[27] Time cost=7.902
INFO:root:Epoch[28] Batch [200] Speed: 2797.25 samples/sec      Train-accuracy=0.007743
INFO:root:Epoch[28] Batch [400] Speed: 2768.36 samples/sec      Train-accuracy=0.008020
INFO:root:Epoch[28] Resetting Data Iterator
INFO:root:Epoch[28] Time cost=7.863
INFO:root:Epoch[29] Batch [200] Speed: 2752.41 samples/sec      Train-accuracy=0.007743
INFO:root:Epoch[29] Batch [400] Speed: 2723.90 samples/sec      Train-accuracy=0.008020
INFO:root:Epoch[29] Resetting Data Iterator
INFO:root:Epoch[29] Time cost=8.005
INFO:root:Epoch[30] Batch [200] Speed: 2772.16 samples/sec      Train-accuracy=0.007743
INFO:root:Epoch[30] Batch [400] Speed: 2768.29 samples/sec      Train-accuracy=0.008020
INFO:root:Epoch[30] Resetting Data Iterator
INFO:root:Epoch[30] Time cost=7.904
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

Maybe I am not familiar with how to write the DataIter.Could someone tell me what's the problem I meet.Thank you so much.

**Comment by piiswrong** [comment]
Please post a minimal reproducing script

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
